File: Python/automorphic_no.py
#A number whose square ends with the number itself
# like 5--> 5^2 = 25;

def findAuto(n):
    auto = False
    m = n**2
    s = 0
    c = 1
    while m != 0:
        s += (m%10)* c
        c *= 10
        m //= 10
        if s == n:
            auto = True
            break
    return(auto)

#__main__
Q = ''
while Q.lower() != 'y':
    print(findAuto(int(input("Enter a number: "))))
    Q = input("Exit(y/n)")

File: C++/automorphic_no.cpp
// A number whose square ends with the number itself
//  like 5--> 5^2 = 25;

#include<iostream>
#include<cmath>
using namespace std;

bool findAuto(int a) {
    int b, t, s, c;
    bool r = false;
    s = 0;
    c = 1;
    b = a*a;
    while (b != 0)
    {
        t = b%10;
        s = s + t * c;
        c = c * 10;
        b = floor(b/10);
        if (s == a) {
            r = true;
            break;
        }
        cout<<s<<endl;
        return r;
    }
    
}

int main() {
    int x;
    cout<<"Enter a number: "<<endl;
    cin>>x;
    if (findAuto(x)==true) {
        cout<<"True"<<endl;
    }
    else {
        cout<<"False"<<endl;
    }
    return 0;
}